### PR TITLE
Remove haproxy from deployments

### PR DESCRIPTION
**Pull request by sgnn7**
We had issues with haproxy using round-robin on all the master servers
even though some of them were not available. By making this change, we
are both removing an unnecessary part of the cluster (haproxy) as well as
fixing a bug in the deployments.

### CODE REVIEW

File: kubernetes/conjur-cluster-service.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+apiVersion: v1
+kind: Service
```
Comment (jtuttle) [inline kubernetes/conjur-cluster-service.yaml]: Just one question: what's this service for? Why are there two?

Comment (sgnn7) [reply kubernetes/conjur-cluster-service.yaml]: LoadBalancer is external only. the other one is for internal routing. (they're somewhat mutually exclusive).

Comment (jtuttle) [reply kubernetes/conjur-cluster-service.yaml]: Mind explaining that in more detail? I'm not really sure what you mean by "external only".

Comment (izgeri) [reply kubernetes/conjur-cluster-service.yaml]: I can see that one has `type: LoadBalancer` - can the other one be explicit about type too to be super clear? there's evidently a default type, but I don't know what it is off the top of my head.

Comment (sgnn7) [reply kubernetes/conjur-cluster-service.yaml]: Added

**Pull request merged**